### PR TITLE
Remove 'kill' alias for Stop-Process cmdlet on Unix

### LINKED ISSUE #10036: kill still an alias of Stop-Process on *nix
Just tried `kill` on macOS and it's an alias of `Stop-Process`. I thought all the aliases masking native utils were taken out?

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Fix #10036

Remove 'kill' alias for Stop-Process cmdlet.

## PR Context

PowerShell Committee approved the breaking change in #10036.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 1]
```diff
@@ -4616,7 +4616,6 @@ internal static SessionStateAliasEntry[] BuiltInAliases
                     new SessionStateAliasEntry("dir", "Get-ChildItem", string.Empty, AllScope),
                     new SessionStateAliasEntry("echo", "Write-Output", string.Empty, AllScope),
                     new SessionStateAliasEntry("fc", "Format-Custom", string.Empty, ReadOnly),
-                    new SessionStateAliasEntry("kill", "Stop-Process"),
```
Comment (rjmholt) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: This should maybe be moved up to the `!UNIX` part instead

File: test/powershell/engine/Basic/DefaultCommands.Tests.ps1 [hunk 2]
```diff
@@ -94,7 +94,7 @@ Describe "Verify approved aliases list" -Tags "CI" {
 "Alias",        "ise",                              "powershell_ise.exe",               $($FullCLR                               ),     "ReadOnly",             "",                     ""
 "Alias",        "iwmi",                             "Invoke-WMIMethod",                 $($FullCLR                               ),     "ReadOnly",             "",                     ""
 "Alias",        "iwr",                              "Invoke-WebRequest",                $($FullCLR -or $CoreWindows -or $CoreUnix),     "ReadOnly",             "",                     ""
-"Alias",        "kill",                             "Stop-Process",                     $($FullCLR -or $CoreWindows -or $CoreUnix),     "",                     "",                     ""
+"Alias",        "kill",                             "Stop-Process",                     $($FullCLR                               ),     "",                     "",                     ""
```
Comment (rjmholt) [inline test/powershell/engine/Basic/DefaultCommands.Tests.ps1]:
```suggestion
"Alias",        "kill",                             "Stop-Process",                     $($FullCLR -or $CoreWindows              ),     "",                     "",                     ""
```

**Review by rjmholt — COMMENTED** [review]
I quickly checked the other utils, and the only other candidates that might be worth considering for this same move are `echo` and `lp`. I presume `echo` was already considered, since it's fairly obvious. But `lp` seems like a safe one since it's an alias for `Out-Printer`, which isn't in PS 7 I don't think. I imagine `lp` itself isn't much used, but I can guarantee that `Out-Printer` isn't being used on *nix.

File: src/System.Management.Automation/engine/InitialSessionState.cs (hunk 1 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/InitialSessionState.cs]: Fixed.

File: test/powershell/engine/Basic/DefaultCommands.Tests.ps1 (hunk 2 above)
Comment (iSazonov) [reply test/powershell/engine/Basic/DefaultCommands.Tests.ps1]: Fixed

**Comment by iSazonov** [comment]
'lp' is only in Windows PowerShell.

**Review by rjmholt — APPROVED** [review]

**Review by rjmholt — APPROVED** [review]

**Pull request merged**